src/Native/System.Native/pal_io.cpp

```diff
@@ -443,3 +464,93 @@ int32_t MksTemps(char* pathTemplate, int32_t suffixLength)
 {
     return mkstemps(pathTemplate, suffixLength);
 }
+
+static
+void ConvertPalPollFDToNative(const PollFD& pal, pollfd& native)
+{
+    native.fd = pal.FileDescriptor;
+    native.events = pal.EventsToPollFor;
+    native.revents = pal.ReceivedEvents;
+}
+
+static
+void ConvertNativePollfdToPal(const pollfd& native, PollFD& pal)
+{
+    pal.FileDescriptor = native.fd;
+    pal.EventsToPollFor = native.events;
+    pal.ReceivedEvents = native.revents;
+}
+
+extern "C"
+int32_t Poll(PollFD* pollData, uint32_t numberOfPollFds, int32_t timeout)
+{
+    assert(numberOfPollFds <= 2);
```

Comment:
Make this a real check, otherwise it will buffer overflow in release if somebody passes 3.